packages/core/src/errors/APIError.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { AmplifyError } from './AmplifyError';
+import { AmplifyErrorParams } from '../types/errors';
+
+/**
+ * The type of an unwrapped HTTP response causing the given API error.
+ * @internal
+ */
+export interface ApiErrorResponse {
+	statusCode: number;
+	headers: Record<string, string>;
+	body?: string;
+}
+
+/**
+ * The constructor parameters for an API error.
+ * @internal
+ */
+export interface ApiErrorParams extends AmplifyErrorParams {
+	response?: ApiErrorResponse;
+}
+
+/**
+ * Error class for errors that associated with unsuccessful HTTP responses.
+ * It's throw by API category REST API handlers and GraphQL query handlers for now.
+ */
+export class ApiError extends AmplifyError {
+	private readonly _response?: ApiErrorResponse;
+
+	/**
+	 * The unwrapped HTTP response causing the given API error.
+	 */
+	public get response(): ApiErrorResponse | undefined {
```

Comment:
nit: public by default

```suggestion
	get response(): ApiErrorResponse | undefined {
```